### PR TITLE
vlc-bin: init at 3.0.20

### PR DESCRIPTION
**Pull request by AndersonTorres**
From https://github.com/NixOS/nixpkgs/pull/209086

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
@ofborg build vlc-bin-arm64 vlc-bin-intel64 vlc-bin-universal

**Comment by AndersonTorres** [comment]
@ofborg build vlc-bin-macos-arm64 vlc-bin-macos-intel64 vlc-bin-macos-universal

**Comment by AndersonTorres** [comment]
@ofborg build vlc-bin-macos-arm64 vlc-bin-macos-intel64 vlc-bin-macos-universal

**Comment by SuperSandro2000** [comment]
Why are we not building from source? From the linked PR it isn't obvious to me.

**Comment by AndersonTorres** [comment]
Because, well, I do not have a Macintosh to play with Darwin idiosyncrasies.

The best I can do for the Nix Darwin users is to repack the binaries provided by upstream.

**Comment by wegank** [comment]
I'd actually prefer a single `vlc-bin` attribute to install the intel64 variant on x86_64-darwin and the arm64 variant on aarch64-darwin, instead of `vlc-bin-macos-{arm64,intel64,universal}`.

**Comment by wegank** [comment]
> Why are we not building from source? From the linked PR it isn't obvious to me.

The author of that PR mentioned that building VLC requires `ibtool` from Xcode, which has no open-source equivalent.

**Comment by AndersonTorres** [comment]
> I'd actually prefer a single `vlc-bin` attribute to install the intel64 variant on x86_64-darwin and the arm64 variant on aarch64-darwin, instead of `vlc-bin-macos-{arm64,intel64,universal}`.

Done. What do you think?

**Comment by pcasaretto** [comment]
Result of `nixpkgs-review pr 291405` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vlc-bin-macos (vlc-bin-macos-universal)</li>
  </ul>
</details>

**Comment by pcasaretto** [comment]
Tested VLC app and working as expected.

**Comment by AndersonTorres** [comment]
@ofborg build vlc-bin

File: pkgs/by-name/vl/vlc-bin-macos/package.nix
Comment (wegank) [inline pkgs/by-name/vl/vlc-bin-macos/package.nix:19]: Is it really necessary to provide a universal one, if there are native versions for both platforms?

File: pkgs/by-name/vl/vlc-bin-macos/package.nix
Comment (wegank) [inline pkgs/by-name/vl/vlc-bin-macos/package.nix:45]: Probably add someone here?

File: pkgs/top-level/all-packages.nix
Comment (wegank) [inline pkgs/top-level/all-packages.nix:35980]: I'd move that logic to `pkgs/by-name/vl/vlc-bin-macos/package.nix` (but is the `-macos` suffix really necessary? The pattern seems unprecedented in Nixpkgs.)

File: pkgs/by-name/vl/vlc-bin-macos/package.nix
Comment (AndersonTorres) [reply pkgs/by-name/vl/vlc-bin-macos/package.nix:19]: I don't know. I included it for completeness.

File: pkgs/top-level/all-packages.nix
Comment (AndersonTorres) [reply pkgs/top-level/all-packages.nix:35980]: I wanted to make it clear that it is a Macos-only binary.

File: pkgs/by-name/vl/vlc-bin-macos/package.nix
Comment (AndersonTorres) [reply pkgs/by-name/vl/vlc-bin-macos/package.nix]: Since I am not a Darwin user, and  I don't know anyone interested, I am putting this package to adoption.

Comment (SuperSandro2000) [reply pkgs/by-name/vl/vlc-bin-macos/package.nix]: If we don't find someone for this, we shouldn't merge this PR.

Comment (pcasaretto) [reply pkgs/by-name/vl/vlc-bin-macos/package.nix]: I'd gladly maintain it. 😃

Comment (AndersonTorres) [reply pkgs/by-name/vl/vlc-bin-macos/package.nix]: I will add you to maintainers set, then!

**Comment by AndersonTorres** [comment]
I think it is now OK!

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 291405` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vlc-bin</li>
    <li>vlc-bin-universal</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 291405` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vlc-bin</li>
    <li>vlc-bin-universal</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
@ofborg build vlc-bin vlc-bin.passthru.tests

File: pkgs/by-name/vl/vlc-bin/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/vl/vlc-bin/package.nix:58]: Please just list the attrs as a list of strings, There should be only really two here anyway.

**Review by pcasaretto — APPROVED** [review]

File: maintainers/maintainer-list.nix
Comment (pcasaretto) [inline maintainers/maintainer-list.nix:14859]:
```suggestion
    name = "Paulo Casaretto";
```
A small request.

**Review by pcasaretto — APPROVED** [review]

File: maintainers/maintainer-list.nix
Comment (AndersonTorres) [reply maintainers/maintainer-list.nix]: As you wish!

**Pull request merged**
